src/nrniv/nrncore_write.cpp

```diff
@@ -238,17 +238,26 @@ static void part2(const char* path) {
   }
 
   // filename data might have to be collected at hoc level since
-  // pc.nrnbbcore_write might be called
+  // pc.nrncore_write might be called
   // many times per rank since model may be built as series of submodels.
-  if (ifarg(2)) {
+  if (ifarg(2) && hoc_is_object_arg(2) && is_vector_arg(2)) {
+    // Legacy style. Interpreter collects groupgids and writes files.dat
     Vect* cgidvec = vector_arg(2);
     vector_resize(cgidvec, nrn_nthread);
     double* px = vector_vec(cgidvec);
     for (int i=0; i < nrn_nthread; ++i) {
       px[i] = double(cgs[i].group_id);
     }
   }else{
-    write_nrnthread_task(path, cgs);
+    bool append = false;
+    if (ifarg(2)) {
+      if (hoc_is_double_arg(2)) {
+        append = (*getarg(2) != 0);
+      }else{
+        hoc_execerror("Second arg must be Vector or double.", NULL);
```

Comment:
Just curious : why it could be vector?

Reply:
The legacy prototype was to fill the second arg hoc Vector with gidgroup values that had to be collected for user creation and writing a files.dat file. In principle that could be restricted to the pc.nrnbbcore_write method name (right now  nrnbbcore_write and nrncore_write are synonyms that go to the same internal nrncore_write function.